src/metabase/lib/util.cljc

```diff
@@ -523,3 +523,23 @@
           ;; subvec holds onto references, so create a new vector
           (update :stages (comp #(into [] %) subvec) 0 (inc (canonical-stage-index query stage-number))))
       new-query)))
+
+(defn fresh-uuids
```

Comment:
Another new util. We need this when expanding legacy Metrics and Segments, because a query might have the same one more than once e.g. if you join the same source query twice.